walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,76 @@
 
 package com.google.sps;
 
-import java.util.Collection;
+import java.util.*;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    int duration = (int)request.getDuration();
+
+    // If request is longer than length of day, then there would never be any options
+    if (duration > TimeRange.WHOLE_DAY.duration()) {
+      return Arrays.asList();
+    }
+
+    // Seperate relevant TimeRanges from events, put into ArrayList and sort by ascending meeting start time
+    ArrayList<TimeRange> meetings = new ArrayList<TimeRange>();
```

Comment:
Preferred form is:
List<TimeRange> meetings = new ArrayList<>();
Why "List"? ArrayList is an implementation specific and List is more usage-oriented form.
Why no "TimeRange"? Simpler and avoids redundancy.

Reply:
done!